cmd/kube-apiserver/app/server.go

```diff
@@ -583,12 +591,46 @@ func BuildAuthenticator(s *options.ServerRunOptions, EgressSelector *egressselec
 	if EgressSelector != nil {
 		egressDialer, err := EgressSelector.Lookup(egressselector.Master.AsNetworkContext())
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
 		authenticatorConfig.CustomDial = egressDialer
 	}
 
-	return authenticatorConfig.New()
+	request, securityDefinitions, err := authenticatorConfig.New()
```

Comment:
seems like this method could return a slice of poststarthooks to add.

Reply:
given the poststarthooks must have unique names, wouldn't coupling it to it's `New` method restrict usage of that object?